wordpress-mobile/WordPress-iOS
# Save for later: toggle button in reader tiles

Implements #9247 

**This PR does not have a milestone yet**. I definitely don't want to get it into the rush for the 10.0 release.

This is an early PR to gather feedback on the general direction it is taking. 

![simulator screen shot - iphone 8 - 2018-05-07 at 12 50 24](https://user-images.githubusercontent.com/2722505/39685458-52955334-51f5-11e8-9c01-ab07ccc3783b.png)

This PR adds a toggle button to the tiles in the reader stream, in the place occupied until now for the Share button. As part of the PR, I have also added some stubbed methods in the `ReaderPostService` and the `ReaderPost` object. The implementation if the service will have to be attacked [here](https://github.com/wordpress-mobile/WordPress-iOS/issues/9265), and the implementation of the model object [here](https://github.com/wordpress-mobile/WordPress-iOS/issues/9227).

After some discussion about the way we want the data model to be, in this PR we diverge from the original plan by embracing a boolean flag in `ReaderPost` instead of adding a separate model object to model posts saved for later.


To test:
1. Deactivate the feature flag
2. Navigate to reader. The Share button should be visible
3. Activate the feature flag
4. Navigate to reader. The bookmark button should be visible instead of the Share button
5. Tapping the bookmark button should make it change color (to blue). The `ReaderPostService` should log a message in the console (to be removed when it is actually implemented)
